reactive/calico.py

```diff
@@ -752,3 +756,16 @@ def get_network(cidr):
 def get_networks(cidrs):
     '''Convert a comma-separated list of CIDRs to a list of networks.'''
     return [get_network(cidr) for cidr in cidrs.split(',')]
+
+
+def is_rpf_config_mismatched():
+    # This check should only apply on focal and above
+    release = lsb_release()
+    release_number = float(release['DISTRIB_RELEASE'])
+    if release_number >= 20.04:
+        with open('/proc/sys/net/ipv4/conf/all/rp_filter') as f:
+            rp_filter = int(f.read())
+        ignore_loose_rpf = charm_config('ignore-loose-rpf')
+        if rp_filter != 1 and not ignore_loose_rpf:
```

Comment:
the config states `ignore_loos_rfp` is only used when `rp_filter` is set to 2.
* https://github.com/charmed-kubernetes/layer-calico/blob/main/config.yaml#L159-L164

but this code says there's a mismatch when `rp_filter != 1`.   does this invalidate either one?

* config says: `ignore_loop_rpf` should only be `True` if `rp_filter==2`
* this check says: mismatch if `ignore_loop_rpf` is `False` and `rp_filter in (0, 2)`

> rp_filter
> 0 No source address validation is performed and any packet is forwarded to the destination network
> 1 Strict Mode as defined in RFC 3704. Each incoming packet to a router is tested against the routing table and if the interface that the packet is received on is not the best return path for the packet then the packet is dropped.
> 2 Loose mode as defines in RFC 3704 Loose Reverse Path. Each incoming packet is tested against the route table and the packet is dropped if the source address is not routable through any interface. The allows for asymmetric routing where the return path may not be the same as the source path

Reply:
https://github.com/kubernetes-sigs/kind/issues/891

>  Calico requires net.ipv4.conf.all.rp_filter to be set to 0 or 1. If you require loose RPF and you are not concerned about spoofing, this check can be disabled by setting the IgnoreLooseRPF configuration parameter to 'true'.


Shouldn't this check just be this:
```python
def is_rpf_config_mismatched():
    with open('/proc/sys/net/ipv4/conf/all/rp_filter') as f:
            rp_filter = int(f.read())
    ignore_loose_rpf = charm_config('ignore-loose-rpf')
    if rp_filter == 2 and not ignore_loose_rpf:
        # calico says this is invalid
        # https://github.com/kubernetes-sigs/kind/issues/891
        return True  
    return False
```

Reply:
you are probably right in that we shouldnt care about the OS version, the reason I added that was that was what field specified in their suggestion in the original bug. But yeah I dont see a point in checking if all we care about is it being 2 or not

Reply:
newest commit removes the OS check and simply checks for rp filter being 2 and ignore_loose_rpf being false as you suggested